scripts/generate_ipfs_hashes.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""
+This script generates the IPFS hashes for all packages.
+
+This script requires that you have IPFS installed:
+- https://docs.ipfs.io/guides/guides/install/
+"""
+
+import collections
+import csv
+import os
+import shutil
+import signal
+import subprocess  # nosec
+import sys
+import time
+from typing import Dict
+
+import ipfshttpclient
+
+from aea.helpers.ipfs.base import IPFSHashOnly
+
+AUTHOR = "fetchai"
+CORE_PATH = "aea"
+CORE_PACKAGES = {"connections": "stub", "protocols": "default", "skills": "error"}
+PACKAGE_PATH = "packages/fetchai"
+PACKAGE_TYPES = ["agents", "connections", "protocols", "skills"]
+PACKAGE_HASHES_PATH = "packages/hashes.csv"
+
+
+def ipfs_hashing(
+    package_hashes: Dict[str, str],
+    target_dir: str,
+    package_type: str,
+    package_name: str,
+    ipfs_hash_only: IPFSHashOnly,
+):
+    """Hashes a package and its components."""
+    print("Processing package {} of type {}".format(package_name, package_type))
+
+    if package_type != "agents":
+        # hash inner components (all `.py` files)
+        result_list = client.add(target_dir, pattern="*.py")
+
+        fingerprints_dict = {}  # Dict[str, str]
+        # get hashes of all `.py` files
+        for result_dict in result_list:
+            if package_name == result_dict["Name"]:
+                continue
+            if not result_dict["Name"][-3:] == ".py":
+                continue
+            key = result_dict["Name"].replace(package_name + "/", "", 1)
+            fingerprints_dict[key] = result_dict["Hash"]
+
+        # confirm ipfs only generates same hash:
+        for file_name, ipfs_hash in fingerprints_dict.items():
+            path = os.path.join(target_dir, file_name)
+            ipfsho_hash = ipfs_hash_only.get(path)
+            if ipfsho_hash != ipfs_hash:
+                print("WARNING, hashes don't match for: {}".format(path))
+
+        # update fingerprints
+        file_name = package_type[:-1] + ".yaml"
```

Comment:
I think this part could be simplified. I'd like to make some suggestions if you don't mind by opening another PR.

Reply:
For instance, we could update the dictionary, rather than working with the content of the file as string.

Reply:
I am sure there is much room for improvement, since it's just a script I did not pay much attention to making it neat. Feel free to improve it.